src/main/java/org/mule/service/http/impl/service/server/ContextHttpServerFactoryAdapter.java

```diff
@@ -19,11 +19,19 @@
  */
 public class ContextHttpServerFactoryAdapter implements HttpServerFactory {
 
+  private final String parentContext;
   private final String context;
   private final ContextHttpServerFactory delegate;
 
   public ContextHttpServerFactoryAdapter(String context, ContextHttpServerFactory delegate) {
     this.context = context;
+    this.parentContext = null;
+    this.delegate = delegate;
+  }
+
+  public ContextHttpServerFactoryAdapter(String context, String parentContext, ContextHttpServerFactory delegate) {
```

Comment:
make `parentContext` an `Optional` and remove/deprecate the previous constructor